From cdba200ab2e4314bb9f200724ad72e85f7179550 Mon Sep 17 00:00:00 2001
From: Dmitriy Aleksandrov <goodmobiledevices@gmail.com>
Date: Mon, 10 May 2021 11:38:19 +0300
Subject: [PATCH] board: nvidia: b3450-0000: setup correct cboot dtb address
 for setup ethernet mac address.

* At startup u-boot get the info from cboot about which address its DTB is in, 
but it transmits the Kernel DTB address, which does not contain information 
about the ethernet mac address. Setup cboot dtb address manualy.

Signed-off-by: Dmitriy Aleksandrov <goodmobiledevices@gmail.com>
---
 board/nvidia/p3450-0000/p3450-0000.c | 2 +-
 arch/arm/mach-tegra/cboot.c | 3 ++-
 2 file changed, 3 insertion(+), 2 deletion(-)

diff --git a/board/nvidia/p3450-0000/p3450-0000.c b/board/nvidia/p3450-0000/p3450-0000.c
index e6b66966..c00f2d94 100644
--- a/board/nvidia/p3450-0000/p3450-0000.c
+++ b/board/nvidia/p3450-0000/p3450-0000.c
@@ -84,7 +84,7 @@ int tegra_pcie_board_init(void)
 
 static void ft_mac_address_setup(void *fdt)
 {
-	const void *cboot_fdt = (const void *)cboot_boot_x0;
+	const void *cboot_fdt = (const void *)0x83000000;
 	uint8_t mac[ETH_ALEN], local_mac[ETH_ALEN];
 	const char *path;
 	int offset, err;
diff --git a/arch/arm/mach-tegra/cboot.c b/arch/arm/mach-tegra/cboot.c
index bb46fb23..5bcc7866 100644
--- a/arch/arm/mach-tegra/cboot.c
+++ b/arch/arm/mach-tegra/cboot.c
@@ -595,6 +595,7 @@ static char *cboot_get_bootargs(const void *fdt)
 int cboot_late_init(void)
 {
 	const void *fdt = (const void *)cboot_boot_x0;
+	const void *bfdt = (const void *)0x83000000;
 	uint8_t mac[ETH_ALEN];
 	char *bootargs;
 	int err;
@@ -607,7 +608,7 @@ int cboot_late_init(void)
 	set_fdt_addr();
 
 	/* Ignore errors here; not all cases care about Ethernet addresses */
-	err = cboot_get_ethaddr(fdt, mac);
+	err = cboot_get_ethaddr(bfdt, mac);
 	if (!err) {
 		void *blob = (void *)gd->fdt_blob;
 
-- 
2.25.1

